core/storage/recommendations/gae_models.py

```diff
@@ -37,6 +37,23 @@ class ExplorationRecommendationsModel(
     recommended_exploration_ids = ndb.StringProperty(
         repeated=True, indexed=False)
 
+    @staticmethod
+    def get_deletion_policy():
+        """Exploration recommendations are deleted only if the corresponding
+        exploration is not public.
+        """
+        return base_models.DELETION_POLICY.KEEP_IF_PUBLIC
+
+    @staticmethod
+    def has_reference_to_user_id(_):
+        """ExplorationRecommendationsModel doesn't reference any user_id
+        directly.
+
+        Returns:
+            bool. Whether the models for user_id exists.
```

Comment:
Whether any models refer to the given user ID.

(Ditto elsewhere.)

Reply:
Done.